src/xb-database-manager.c

```diff
@@ -721,6 +722,90 @@ parse_default_op (XapianQueryParser *qp, const gchar *str, GError **error)
   return TRUE;
 }
 
+static gboolean
+parse_query_flags (const gchar *str,
+                   XapianQueryParserFeature *flags_ptr,
+                   GError **error)
+{
+  XapianQueryParserFeature flags = 0;
+  gchar **v = g_strsplit (str, ",", -1), **iter;
+  for (iter = v; *iter != NULL; iter++)
+    {
+      if (g_str_equal (*iter, "boolean"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_BOOLEAN;
+        }
+      else if (g_str_equal (*iter, "phrase"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_PHRASE;
+        }
+      else if (g_str_equal (*iter, "lovehate"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_LOVEHATE;
+        }
+      else if (g_str_equal (*iter, "boolean-any-case"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_BOOLEAN_ANY_CASE;
+        }
+      else if (g_str_equal (*iter, "wildcard"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_WILDCARD;
+        }
+      else if (g_str_equal (*iter, "pure-not"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_PURE_NOT;
+        }
+      else if (g_str_equal (*iter, "partial"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_PARTIAL;
+        }
+      else if (g_str_equal (*iter, "spelling-correction"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_SPELLING_CORRECTION;
+        }
+      else if (g_str_equal (*iter, "synonym"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_SYNONYM;
+        }
+      else if (g_str_equal (*iter, "auto-synonyms"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_AUTO_SYNONYMS;
+        }
+      else if (g_str_equal (*iter, "auto-multiword-synonyms"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_AUTO_MULTIWORD_SYNONYMS;
+        }
+      else if (g_str_equal (*iter, "cjk-ngram"))
+        {
+#ifdef XAPIAN_QUERY_PARSER_FEATURE_CJK_NGRAM
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_CJK_NGRAM;
+#else
+          // Not wrapped by older xapian-glib.
+          g_set_error (error, XB_ERROR,
+                       XB_ERROR_INVALID_PARAMS,
+                       "Query parser flag 'cjk-ngram' not supported by xapian-glib in use.");
+          g_strfreev (v);
+          return FALSE;
+#endif
+        }
+      else if (g_str_equal (*iter, "default"))
+        {
+          flags |= XAPIAN_QUERY_PARSER_FEATURE_DEFAULT;
+        }
+       else
```

Comment:
Fixed.

Given the trivial changes, I've force-pushed an amended branch for convenience of merging.